### PR TITLE
Check for list index within range

### PR DESCRIPTION
**Pull request by ZER-0-NE**
List Index will be out of range for sys.argv[1]. Adding a check for that.

### CODE REVIEW

**Comment by RyuzakiKK** [comment]
Thanks for the patch.
It should be better to raise a ValueError with a short explanation of what is missing rather than silently switching to sys.argv[0].

**Comment by ZER-0-NE** [comment]
I have added an explanation of the error. Please check it. @RyuzakiKK

**Comment by RyuzakiKK** [comment]
Maybe I wasn't clear enough. What I said was that we want to use the `sys.argv[1]` (the first argument passed by the user) and that we don't care about `sys.argv[0]` (the name of the file executed).

So you can raise the `ValueError` right after the arguments length check fails (the `else`).

**Comment by ZER-0-NE** [comment]
Updated as required. @RyuzakiKK

File: keysign/QRCode.py [hunk 1]
```diff
@@ -317,5 +318,8 @@ def on_released(widget, event):
 if __name__ == '__main__':
     import sys
     logging.basicConfig(level=logging.DEBUG)
-    data = sys.argv[1]
+    if len(sys.argv) > 1:
+	    data = sys.argv[1]
```
Comment (RyuzakiKK) [inline keysign/QRCode.py]: Be careful with the indentation. With Python we follow as much as possible the PEP 8, so we use 4 spaces.
Here you mixes tabs and spaces.

File: keysign/QRCode.py [hunk 2]
```diff
@@ -317,5 +318,8 @@ def on_released(widget, event):
...
+    else:
+            raise ValueError("Not Enough Arguments passed as data for the QR code encoding")
```
Comment (RyuzakiKK) [inline keysign/QRCode.py]: Indented too much. 12 spaces instead of 8.

File: keysign/QRCode.py (hunk 1 above)
Comment (ZER-0-NE) [reply keysign/QRCode.py]: That would be my editor.

File: keysign/QRCode.py (hunk 2 above)
Comment (ZER-0-NE) [reply keysign/QRCode.py]: @RyuzakiKK Done. Sorry about that.

**Comment by ZER-0-NE** [comment]
Finally Done! (I think) @RyuzakiKK Earlier I used to use tabs for writing my code. Will be careful about this here.

**Comment by RyuzakiKK** [comment]
I suggest you to use an IDE like PyCharm, it will simplify your life a lot. But also very simple text editors like Gedit can be configured to insert 4 spaces when you press tab.

**Comment by muelli** [comment]
@ZER-0-NE thanks for the contribution.
I'm not going to merge seven commits for this change. Maybe I would, but only if they have a descriptive message each. During the next rebase session hunting down a bug, I want to be able to understand why each change was made. " Adding required changes. " is not helping me there.

I'm usually not a big fan of squashing commits or altering history, but in this case I ask you to make a single commit with a nice commit message. Or maybe actually two, because you seem to be changing two things. The Gdk import and the sys.args.
You can force-push into your proposed branch.

File: keysign/QRCode.py [hunk 3]
```diff
@@ -317,5 +318,8 @@ def on_released(widget, event):
 if __name__ == '__main__':
     import sys
     logging.basicConfig(level=logging.DEBUG)
-    data = sys.argv[1]
+    if len(sys.argv) > 1:
```
Comment (muelli) [inline keysign/QRCode.py]: While you're at it, can you make it a `try..catch` rather than a length test? Read PEP-020 for the Zen of Python :)

Comment (ZER-0-NE) [reply keysign/QRCode.py]: That's some wise principles :)
Thank you. :D

**Comment by ZER-0-NE** [comment]
@muelli Done!

**Comment by muelli** [comment]
cool. thanks.
I've reworked your commits a little bit, mainly to remove the whitespace noise.

de9a7b6ae0df7e1a23e7ef8a3f3f61a5d05af677

**Pull request closed without merging**